x/ccv/consumer/keeper/relay.go

```diff
@@ -103,24 +104,25 @@ func (k Keeper) UnbondMaturePackets(ctx sdk.Context) error {
 }
 
 // SendSlashPacket sends a slash packet containing the given validator data and slashing info
-func (k Keeper) SendSlashPacket(ctx sdk.Context, validator abci.Validator, valsetUpdateID uint64, slashFraction, jailedUntil int64) {
+func (k Keeper) SendSlashPacket(ctx sdk.Context, validator abci.Validator, valsetUpdateID uint64, infraction stakingtypes.InfractionType) {
+	consAddr := sdk.ConsAddress(validator.Address)
+	downtime := infraction == stakingtypes.Downtime
 
 	// return if an outstanding downtime request is set for the validator
-	consAddr := sdk.ConsAddress(validator.Address)
-	if k.OutstandingDowntime(ctx, consAddr) { // TODO: add to condition if the slash is for downtime
+	if downtime && k.OutstandingDowntime(ctx, consAddr) {
 		return
 	}
 
 	// construct slash packet data
-	packetData := ccv.NewSlashPacketData(validator, valsetUpdateID, slashFraction, jailedUntil)
+	packetData := ccv.NewSlashPacketData(validator, valsetUpdateID, infraction)
 
 	// check that provider channel is established
 	// if not, append slashing packet to pending slash requests
 	channelID, ok := k.GetProviderChannel(ctx)
 	if !ok {
 		k.AppendPendingSlashRequests(ctx, types.SlashRequest{
 			Packet:   &packetData,
-			Downtime: true}, // TODO Simon: add double-signing check
+			Downtime: downtime},
```

Comment:
It's odd here to use a boolean to determine what type of slash it is, where everywhere else you use an enum.

I suggest that this gets changed to be the enum value as well.

Reply:
Agree. `PendingSlashRequest` msg definition updated accordingly.